src/scripts/main.js

```diff
@@ -1,3 +1,27 @@
 'use strict';
 
-// write your code here
+const populationElements = document.querySelectorAll('span.population');
+
+const populations = Array.from(populationElements)
+  .map((element) => element.textContent.trim())
+  .map(text => text.replace(/,/g, ''))
+  .filter((text) => !isNaN(text && text !== ''))
```

Comment:
The filter condition `!isNaN(text && text !== '')` is incorrect. The logical AND (`&&`) operator will return the second operand if the first is truthy, which is not the intended behavior here. Instead, you should check `text !== '' && !isNaN(text)` to ensure that non-empty strings are checked for being a number.